src/compile/data/optimize.ts

```diff
@@ -47,35 +47,34 @@ function runOptimizer(optimizer: BottomUpOptimizer | TopDownOptimizer, nodes: Da
       return optimizer.run(node);
     }
   });
-  return flags.some(isTrue) || flag;
+  return flags.some(isTrue);
 }
 
 function optimizationDataflowHelper(dataComponent: DataComponent, model: Model) {
   let roots = dataComponent.sources;
-  let mutatedFlag = false;
+  const mutatedFlags: boolean[] = [];
 
-  // mutatedFlag should always be on the right side otherwise short circuit logic might cause the mutating method to not execute
-  mutatedFlag = runOptimizer(new optimizers.RemoveUnnecessaryNodes(), roots, mutatedFlag);
+  mutatedFlags.push(runOptimizer(new optimizers.RemoveUnnecessaryNodes(), roots));
```

Comment:
Why not just `mutatedFlag = mutatedFlag || runOptimizer(new optimizers.RemoveUnnecessaryNodes(), roots)`? Then we avoid creating an array.

Reply:
That's the pattern I wanted to avoid.
Javascript has short circuit logic, so the 2nd part never runs if the 1st is true.
```
> console.log(true || 1/0 );
true
```

Reply:
Oh, good point. I totally missed that.